src/framework/global/types/string.cpp

```diff
@@ -722,6 +722,9 @@ StringList String::split(const std::regex& re, SplitBehavior behavior) const
 
 String& String::replace(const String& before, const String& after)
 {
+    if (before == after) {
+        return *this;
```

Comment:
Actually it's only in the case that before 'before' and 'after' are empty strings that it hangs, but this is  a worthwhile optimisation anyway.